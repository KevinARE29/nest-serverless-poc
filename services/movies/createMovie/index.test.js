/**
 * @group unit
 */

const eventData = require('./exampleData.json')
const { handler } = require('./index')

describe('createMovie', () => {
  it('should return a 400 error if the name is not defined', async () => {
    const body = JSON.parse(eventData.body)

    const result = await handler({
      ...eventData,
      body: JSON.stringify({ ...body, name: null }),
    })

    expect(result.statusCode).toBe(400)
    expect(JSON.parse(result.body).message).toBe('name must be a string')
  })

  it('should return a 400 error if the duration is not valid', async () => {
    const body = JSON.parse(eventData.body)

    const result = await handler({
      ...eventData,
      body: JSON.stringify({ ...body, duration: '-1' }),
    })

    expect(result.statusCode).toBe(400)
    expect(JSON.parse(result.body).message).toBe(
      'duration must be a positive number',
    )
  })

  it('should return a 400 error if the poster is not defined', async () => {
    const body = JSON.parse(eventData.body)

    const result = await handler({
      ...eventData,
      body: JSON.stringify({ ...body, poster: null }),
    })

    expect(result.statusCode).toBe(400)
    expect(JSON.parse(result.body).message).toBe('poster is required')
  })

  it('should create a movie', async () => {
    const result = await handler(eventData)
    const body = JSON.parse(result.body)

    expect(result.statusCode).toBe(201)
    expect(body).toMatchSnapshot({
      data: {
        uuid: expect.any(String),
        name: expect.any(String),
        duration: expect.any(Number),
        price: expect.any(Number),
        likes: expect.any(Number),
        isActive: expect.any(Boolean),
        poster: expect.any(String),
      },
    })
  })
})
