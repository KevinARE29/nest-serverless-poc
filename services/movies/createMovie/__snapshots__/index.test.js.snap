// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createMovie should create a movie 1`] = `
Object {
  "data": Object {
    "duration": Any<Number>,
    "isActive": Any<Boolean>,
    "likes": Any<Number>,
    "name": Any<String>,
    "poster": Any<String>,
    "price": Any<Number>,
    "synopsis": null,
    "uuid": Any<String>,
  },
}
`;
